### DISCUSSION TITLE
How can I initialize the base class without default constructor?

### DISCUSSION THREAD

**Discussion by shelllet** [microsoft/WindowsAppSDK#1450] [Q&A] [answered]
`idl`
```
[default_interface]
unsealed runtimeclass ActionControl : Microsoft.UI.Xaml.Controls.ContentControl
{
	ActionControl(String key);
}

[default_interface]
runtimeclass Xaml_StartProcess : ActionControl
{
	Xaml_StartProcess(String key);
}
```

`projection class`

```
struct Xaml_StartProcess : Xaml_StartProcessT<Xaml_StartProcess>
{
	Xaml_StartProcess(hstring const & key) : implements_type(key){}
};
```
i try to add implements_type, but
compile error: `illegal member initialization: 'implements_type' is not a base or member`

**Comment by btueffers** [comment, marked as answer]
Hi @shelllet is there any additional information about your question that is specific to XAML? If not, the [Microsoft Q & A](https://docs.microsoft.com/en-us/answers/products/) would be a better place to have this discussion.

**Reply by shelllet** [reply]
thx, may be  [`cppwinrt`](https://github.com/microsoft/cppwinrt)  is a better place.
